vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,673 @@
+      active = true;
+      animfrac = 0.0f;
+      repaintArea();
+    }
+
+    /**
+     * Animator tells us when to update the image.
+     * @param fraction Animator lets us know how far we are through our cycle
+     */
+    @Override
+    public void timingEvent(float fraction) {
+      animfrac = fraction;
+      repaintArea();
+    }
+
+    /**
+     * Animator tells us we're done.
+     */
+    @Override
+    public void end() {
+      active = false;
+      repaintArea();
+    }
+  });
+
+  /**
+   * Start the Flare animation.
+   * @param isLocal - true if this Flare was launched on this client (i.e. by this player). Otherwise, center on
+   *                  the ping location if user preferences are set for centering on opponent moves.
+   */
+  public void startAnimation(final boolean isLocal) {
+    if (map == null) {
+      return; // Means we already mentioned BadModuleData earlier. Now we simply avoid crashing.
+    }
+
+    if (!isLocal) {
+      if (GlobalOptions.getInstance().centerOnOpponentsMove()) {
+        map.centerAt(clickPoint);
+      }
+    }
+
+    animator.stop();
+    animate = pulses > 0 && pulsesPerSec > 0;
+    animator.setRepeatCount(Math.max(pulses, 1));
+    animator.setDuration(1000 / Math.max(pulsesPerSec, 1));
+    animator.start();
+  }
+
+  /**
+   * Draw the Flare at current animation frame
+   * @param g - Graphics
+   * @param map - Map component
+   */
+  public void draw(final Graphics g, final Map map) {
+    if (!active || (clickPoint == null)) {
+      return;
+    }
+    final Graphics2D g2d = (Graphics2D) g;
+    os_scale = g2d.getDeviceConfiguration().getDefaultTransform().getScaleX();
+
+    double diameter = (circleScale ? map.getZoom() : 1.0) * os_scale * circleSize;
+    if (animate) {
+      diameter *= (1.0 - animfrac);
+    }
+
+    if (diameter <= 0.0) {
+      return;
+    }
+
+    // translate the click location for current zoom
+    final Point p = map.mapToDrawing(clickPoint, os_scale);
+
+    // draw a circle around the selected point
+    g2d.setColor(color);
+    g2d.setStroke(new BasicStroke((float)(STROKE * os_scale)));
+    g2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING,
+            RenderingHints.VALUE_ANTIALIAS_ON);
+    g2d.drawOval(
+            (int)(p.x - diameter / 2.0),
+            (int)(p.y - diameter / 2.0),
+            (int)(diameter + 0.5),
+            (int)(diameter + 0.5)
+    );
+  }
+
+  /**
+   * Flare is always drawn above counters
+   * @return true
+   */
+  public boolean drawAboveCounters() {
+    return true;
+  }
+
+  /**
+   * Serializes Flare commands into string form.
+   * @param c Command
+   * @return String version of Flare Command, or null if not a Flare Command.
+   */
+  public String encode(final Command c) {
+    if (c instanceof FlareCommand) {
+      return COMMAND_PREFIX + ((FlareCommand) c).getId() + DELIMITER + ((FlareCommand) c).getClickPoint().x + DELIMITER + ((FlareCommand) c).getClickPoint().y;
```

Comment:
These are the places you say you need `DELIMITER` to be a `String`? I think it would work with `DELIMITER` as a `char` as well.

Reply:
Also, it's dangerous not to use `SequenceEncoder` if any of the items you're encoding are `Strings` which you don't control. E.g., what happens if `getId()` has `DELIMITER` in it?

Reply:
I am certainly not an expert on "why we use the delimiters that we use" in the various SequenceEncoders. I started with a fairly traditional/naive delimiter (e.g. "," or something), and Brent suggested I change that to "\t" or something "for reasons", and then Yan suggested I define as DELIMITER so that I had it all in one place rather than hard-coding it. This seemed like a plausible idea, but when I attempted to do that I discovered that while "usually" the delimiters appear being +'ed into strings e.g. "Thing" + "\t" + "OtherThing", that SequenceDecoder (only) really wanted a char '\t' or whatever. Which then made the whole idea of "making the code less confusing and/or more maintainable" by defining DELIMITER seem farcical unless the same thing could be used in all places, and the easiest candidate for change was letting SequenceDecoder accept a string delimiter. I suppose we could throw an exception if it was other than exactly 1 character long. 

Anyway if you don't like the simplifications I'm happy to go back to hard-coding "\t" or whatever "like most classes do". Or if you do like them, just let me know which way you want it although trying to append `char`'s into the middle of Strings is surprisingly inconvenient in Java so I'm not a fan of that.

Reply:
Okay the char version of DELIMITER "just works". God knows what I was doing. Probably forgot to declare DELIMITER to be char and didn't notice that was the only problem.

Pulled out the special bonus version of SequenceEncoder, which earned me the opportunity to "mergetool" it about 6 consecutive times during the rebase :D